src/plone/api/tests/test_content.py

```diff
@@ -1041,8 +1042,8 @@ def test_find_interface_dict__include_not_query(self):
                 "not": INavigationRoot.__identifier__,
             },
         )
-
-        self.assertEqual(len(brains_all) - len(brains), 1)
+        # Plone Site also implements the IContentish and INavigationRoot interfaces.
+        self.assertEqual(len(brains_all) - len(brains), 2)
```

Comment:
@wesleybl This couples plone.api closely to the version of CMFPlone (i.e. a new version of plone.api will require a new version of CMFPlone, and an old version of plone.api will require an old version of CMFPlone...at least in order to have these tests pass.

To make it less tightly coupled, I think you could add `"not": IPloneSiteRoot.__identifier__` to the query, and continue to check for 1 result. Then the test will work with and without the update to the Plone site interfaces.

Reply:
It seems we are initially getting all objects that provide the IContentish interface and then applying a 'not' query to exclude those that provide the INavigationRoot interface. The assertion should be updated to check that there is one less object in the result compared to the initial count. The updated assertion should look like self.assertEqual(len(brains_all) - len(brains), 1). is that right @davisagli @wesleybl?

Reply:
@Akshat2Jain Oh yeah, comparing the final count to the initial count would also work. Good idea

Reply:
@davisagli good idea! The test `test_find_interface_dict` would have to be fixed too.

> The updated assertion should look like self.assertEqual(len(brains_all) - len(brains), 1)

@Akshat2Jain Plone Site also implements the `INavigationRoot` interface. So a search with `"not": INavigationRoot.__identifier__` will also exclude the Plone Site. So we will have two less.

I think the best option is @davisagli 's suggestion. The only "problem" with this option is that if new content is created in the test class, the test will fail and will have to be fixed. But I don't think that's such a big problem.

Reply:
Should I undo the change then?

Reply:
@Akshat2Jain yes. You should leave the assert with `1` and use `IPloneSiteRoot.__identifier__` in not. The line with `alsoProvides` is no longer needed.

Reply:
I think we could use `plone.app.contenttypes.interfaces.IEvent` in place of `IContentish`, and keep using `INavigationRoot` in the `not`. So the `alsoProvides` would be in `self.training` instead of `self.portal.events`.